src/api/consumer/lib/switch.ts

```diff
@@ -26,51 +26,55 @@ export default async function switchAction(
 
 async function populateSwitchProps(consumer: Consumer, switchProps: SwitchProps) {
   const lanes = await consumer.scope.listLanes();
-  const { laneName, remoteScope } = switchProps;
-  if (remoteScope) {
+  const localLane = lanes.find((lane) => lane.name === switchProps.laneName);
+
+  if (!localLane) {
     await populatePropsAccordingToRemoteLane();
   } else {
     populatePropsAccordingToLocalLane();
   }
 
   async function populatePropsAccordingToRemoteLane() {
+    const remoteLaneId = RemoteLaneId.parse(switchProps.laneName);
+    switchProps.remoteLaneName = remoteLaneId.name;
+    switchProps.laneName = remoteLaneId.name;
+    switchProps.remoteLaneScope = remoteLaneId.scope;
+    switchProps.remoteScope = remoteLaneId.scope;
     // fetch the remote to update all heads
-    const localTrackedLane = consumer.scope.lanes.getLocalTrackedLaneByRemoteName(laneName, remoteScope as string);
-    switchProps.localLaneName = switchProps.newLaneName || localTrackedLane || laneName;
+    const localTrackedLane = consumer.scope.lanes.getLocalTrackedLaneByRemoteName(
+      switchProps.laneName,
+      switchProps.remoteScope as string
+    );
+    switchProps.localLaneName = switchProps.newLaneName || localTrackedLane || switchProps.laneName;
     if (consumer.getCurrentLaneId().name === switchProps.localLaneName) {
       throw new GeneralError(`already checked out to "${switchProps.localLaneName}"`);
     }
     const scopeComponentImporter = ScopeComponentsImporter.getInstance(consumer.scope);
-    const remoteLaneObjects = await scopeComponentImporter.importFromLanes([
-      RemoteLaneId.from(laneName, remoteScope as string),
-    ]);
+    const remoteLaneObjects = await scopeComponentImporter.importFromLanes([remoteLaneId]);
     const remoteLaneComponents = remoteLaneObjects[0].components;
     const laneExistsLocally = lanes.find((l) => l.name === switchProps.localLaneName);
     if (laneExistsLocally) {
-      throw new GeneralError(`unable to checkout to a remote lane ${remoteScope}/${laneName}.
+      throw new GeneralError(`unable to checkout to a remote lane ${switchProps.remoteScope}/${switchProps.laneName}.
     the local lane ${switchProps.localLaneName} already exists, please switch to the local lane first by omitting --remote flag
```

Comment:
This error should now be changed. It should suggest removing the scope-name probably.

Reply:
fixed